core/state_test.go

```diff
@@ -169,7 +171,8 @@ func TestUpdate(t *testing.T) {
 	testDb := pebble.NewMemTest()
 	state := core.NewState(testDb.NewTransaction(true))
 
-	assert.Equal(t, nil, state.Update(coreUpdate))
+	// coreUpdate has no declared contracts
+	assert.Equal(t, nil, state.Update(coreUpdate, emptyDeclaredContracts))
```

Comment:
I think passing nil would work just fine. Please update all the tests accordingly.